client/src/components/AllClubs.tsx

```diff
@@ -1,10 +1,97 @@
-import NavBar from "./NavigationBar";
+import React, { useState } from 'react';
+import NavBar from './NavigationBar';
+import { Link } from 'react-router-dom';
+import '../AllClubs.css'; // Import the CSS file
 
 export default function AllClubs() {
+  const initialClubs = [
+    { name: 'Tennis Club', id: 'tennis' },
+    { name: 'Chess Club', id: 'chess' },
+    { name: 'Reading Club', id: 'reading' },
+    { name: 'Science Club', id: 'science' },
+    { name: 'Art Club', id: 'art' },
+    { name: 'Music Club', id: 'music' },
+    { name: 'Cooking Club', id: 'cooking' },
+    { name: 'Photography Club', id: 'photography' },
+    { name: 'Gaming Club', id: 'gaming' },
+    { name: 'Dance Club', id: 'dance' },
+  ];
+
+  const [clubs, setClubs] = useState(initialClubs);
+  const [joinedClubs, setJoinedClubs] = useState([]);
+  const [query, setQuery] = useState('');
+
+  // Function to handle joining a club
+  const joinClub = (clubId) => {
+    // ... (rest of the joinClub function)
+  };
+
+  // Function to handle leaving a joined club
+  const leaveClub = (clubId) => {
+    // ... (rest of the leaveClub function)
+  };
+
+  // Filter clubs based on query
+  const filteredClubs = clubs.filter((club) =>
+    club.name.toLowerCase().includes(query.toLowerCase())
+  );
+
   return (
-    <div>
+    <div className="my-clubs-container">
       <NavBar />
-      All Clubs
+
+      <div className="total-club-count">
+        Total Clubs Joined: {joinedClubs.length}
+      </div>
+
+      <h2 className="club-heading">All Clubs</h2>
+
+      <input
+        type="text"
+        placeholder="Search clubs..."
+        value={query}
+        onChange={(e) => setQuery(e.target.value)}
+        className="search-input"
+      />
+
+      <table className="clubs-table">
+        <tbody>
+          {filteredClubs.map((club) => (
+            <tr key={club.id}>
+              <td>
+                <Link
+                  to={`/club/${club.id}?clubName=${encodeURIComponent(
+                    club.name
+                  )}`}
+                  className="club-link"
+                >
+                  {club.name}
+                </Link>
+                <button className="join-button" onClick={() => joinClub(club.id)}>
+                  Join
+                </button>
+              </td>
+            </tr>
+          ))}
+          {joinedClubs.map((club) => (
```

Comment:
With the way this is formatted, you're going to have every possible club listed out first (including ones you're in), then you'll have every club you're in, so you'll have duplicate clubs and users will see a "Join" button on clubs they're already in.

IMO you can just remove join buttons from this AllClubs page and move it to the actual specific club page, so users have to click onto a club's page before joining it. Then you can get rid of `joinedClubs`. It'll be a lot cleaner and an easier fix to the above issue